packages/quantic/cypress/integration/recent-queries-list/recent-queries-list.cypress.ts

```diff
@@ -0,0 +1,100 @@
+import {configure, reset} from '../../page-objects/configurator';
+import {InterceptAliases, interceptSearch} from '../../page-objects/search';
+import {RecentQueriesListExpectations as Expect} from './recent-queries-list-expectations';
+import {scope} from '../../reporters/detailed-collector';
+import {
+  clearInput,
+  performSearch,
+  setQuery,
+} from '../../page-objects/actions/action-perform-search';
+
+interface RecentQueriesListOptions {
+  maxLength: number;
+  label: string;
+  isCollapsed: boolean;
+}
+
+function createRandomQueriesList(length: number) {
+  const result = [];
+  for (let index = 0; index < length; index++) {
+    result.push(
+      Math.random()
+        .toString(36)
+        .replace(/[^a-z]+/g, '')
+        .substr(0, 8)
+    );
+  }
+  return result;
+}
+
+describe('quantic-recent-queries-list', () => {
+  const pageUrl = 's/quantic-recent-queries-list';
+  const defaultMaxLength = 10;
+
+  function visitRecentQueries(options: Partial<RecentQueriesListOptions> = {}) {
+    interceptSearch();
+    cy.visit(pageUrl);
+    configure(options);
+  }
+
+  it('should render a placeholder before results have returned', () => {
+    visitRecentQueries();
+
+    Expect.displayPlaceholder(true);
+    Expect.displayQueries(false);
+    Expect.displayEmptyList(true);
+  });
+
+  describe('with default options', () => {
+    it('should work as expected', () => {
+      visitRecentQueries();
+
+      scope('when loading the page', () => {
+        Expect.displayQueries(false);
+        Expect.displayEmptyList(true);
+      });
+
+      scope('when searching new queries', () => {
```

Comment:
We could also add a test to validate that the component displays distinct queries.